### ISSUE TITLE
Unexpected Error when I activate `root`

### ISSUE THREAD

**Issue by toszter** [conda/conda#4038]
1. Listed envs with `conda info --envs`.
```
dato-env                 /Users/toszter/Engineering/machinelearning/anaconda/envs/dato-env
gl-env                   /Users/toszter/Engineering/machinelearning/anaconda/envs/gl-env
root                  *  /Users/toszter/Engineering/machinelearning/anaconda
```
1. Received this unexpected error when I `source activate root`:
```
An unexpected error has occurred.
Please consider posting the following information to the
conda GitHub issue tracker at:

    https://github.com/conda/conda/issues



Current conda install:

               platform : osx-64
          conda version : 4.2.13
       conda is private : False
      conda-env version : 4.2.13
    conda-build version : 0+unknown
         python version : 2.7.12.final.0
       requests version : 2.11.1
       root environment : /Users/toszter/Engineering/machinelearning/anaconda  (writable)
    default environment : /Users/toszter/Engineering/machinelearning/anaconda
       envs directories : /Users/toszter/Engineering/machinelearning/anaconda/envs
          package cache : /Users/toszter/Engineering/machinelearning/anaconda/pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/osx-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/pro/osx-64
                          https://repo.continuum.io/pkgs/pro/noarch
            config file : None
           offline mode : False



`$ /Users/toszter/Engineering/machinelearning/anabackuponda/bin/conda ..checkenv root`




    Traceback (most recent call last):
      File "/Users/toszter/Engineering/machinelearning/anaconda/lib/python2.7/site-packages/conda/exceptions.py", line 479, in conda_exception_handler
        return_value = func(*args, **kwargs)
      File "/Users/toszter/Engineering/machinelearning/anaconda/lib/python2.7/site-packages/conda/cli/main.py", line 94, in _main
        activate.main()
      File "/Users/toszter/Engineering/machinelearning/anaconda/lib/python2.7/site-packages/conda/cli/activate.py", line 108, in main
        shelldict = shells[shell]
    KeyError: 'root'
```

**Comment by njalerikson** [comment]
`activate` is not a conda function so out of the box you cannot run `conda activate ENV`. However if you did run `conda activate ...` it would have given you a different error so I suspect that was a typo.

My guess is that you are using an unsupported shell.

**Comment by toszter** [comment]
@njalerikson - 
```
shared $ > echo -$0
--bash
```

**Comment by njalerikson** [comment]
Well that's just an unfortunate surprise...

After running `source activate root` when you get the above error message, what do you get if you type:

```
echo "[${_SHELL},${EXT},${args}]"
```

**Comment by toszter** [comment]
```
Engineering $ > echo "[${_SHELL},${EXT},${args}]"
[,,]
```

**Comment by toszter** [comment]
I had to move fwd, so I went ahead and did myself a favor, and deleted the installation created by the Dato Launcher... Instead, I removed all conda directories and .bash_profile mods. Then, reinstalled using the v4.0.0 shell script and then `conda update conda` and then `conda update anaconda`. Now, running conda v4.2.13... 

So then I ran `conda create --name super_ml python=2.7` successfully...

...and when that finished, `source activate super_ml`.

and I live happily ever after. No errors. I think what was the culprit was I had multiple conda versions from using an ML library from Dato that I was using.
